### PR TITLE
Montana taxable income

### PR DESCRIPTION
**Pull request by hua7450**
Merge branch 'master' of https://github.com/PolicyEngine/policyengine-us into hua7450/issue3307

### CODE REVIEW

File: policyengine_us/variables/gov/states/mt/tax/income/deductions/itemized/mt_itemzied_deductions_less_salt.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+from policyengine_us.model_api import *
+
+
+class mt_itemized_deductions_less_salt(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Montana itemized deductions excluding SALT"
+    unit = USD
+    definition_period = YEAR
+    adds = ["mt_itemzied_deductions"]
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mt/tax/income/deductions/itemized/mt_itemzied_deductions_less_salt.py:10]: we can delete the adds and subtracts here

File: policyengine_us/variables/gov/states/mt/tax/income/additions/mt_additions.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+from policyengine_us.model_api import *
+
+
+class mt_additions(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Montana additions"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mt/tax/income/additions/mt_additions.py:7]:
```suggestion
    label = "Montana additions to adjusted gross income"
```

File: policyengine_us/variables/gov/states/mt/tax/income/deductions/itemized/mt_itemzied_deductions_less_salt.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mt/tax/income/deductions/itemized/mt_itemzied_deductions_less_salt.py:1]: dont need this file, just use mt_itemzied_deductions

File: policyengine_us/variables/gov/states/mt/tax/income/mt_taxable_income.py [hunk 3]
```diff
@@ -8,3 +8,17 @@ class mt_taxable_income(Variable):
     unit = USD
     definition_period = YEAR
     defined_for = StateCode.MT
+    reference: "https://mtrevenue.gov/wp-content/uploads/dlm_uploads/2023/05/Montana-Idividiual-Income-Tax-Return-Form-2-2022v6.2.pdf#page=1"
+
+    def formula(tax_unit, period, parameters):
+        mt_agi = tax_unit("mt_agi", period)
+        standard_deduction = tax_unit("mt_standard_deduction", period)
+        itemizes = tax_unit("tax_unit_itemizes", period)
+        itemized_deductions_less_salt = tax_unit(
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mt/tax/income/mt_taxable_income.py:17]:
```suggestion
        itemized_deductions = tax_unit(
```

File: policyengine_us/variables/gov/states/mt/tax/income/mt_taxable_income.py [hunk 4]
```diff
@@ -8,3 +8,17 @@ class mt_taxable_income(Variable):
...
+            "mt_itemized_deductions_less_salt", period
+        )
+        deductions = where(
+            itemizes, itemized_deductions_less_salt, standard_deduction
+        )
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mt/tax/income/mt_taxable_income.py:22]:
```suggestion
        # Tax units can claim the larger of the itemized or standard deductions
        deductions = max_(itemized_deductions, standard_deduction)
```

File: policyengine_us/variables/gov/states/mt/tax/income/mt_taxable_income.py [hunk 5]
```diff
@@ -8,3 +8,17 @@ class mt_taxable_income(Variable):
     unit = USD
     definition_period = YEAR
     defined_for = StateCode.MT
+    reference: "https://mtrevenue.gov/wp-content/uploads/dlm_uploads/2023/05/Montana-Idividiual-Income-Tax-Return-Form-2-2022v6.2.pdf#page=1"
+
+    def formula(tax_unit, period, parameters):
+        mt_agi = tax_unit("mt_agi", period)
+        standard_deduction = tax_unit("mt_standard_deduction", period)
+        itemizes = tax_unit("tax_unit_itemizes", period)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mt/tax/income/mt_taxable_income.py:16]:
```suggestion
```

**Review by PavelMakarchuk — APPROVED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**